Sources/AsyncHTTPClient/AsyncAwait/HTTPClientResponse.swift

```diff
@@ -138,6 +138,17 @@ extension HTTPClientResponse {
     }
 }
 
+@available(macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0, *)
+extension HTTPClientResponse {
+    /// Response body as `ByteBuffer`.
+    /// - Parameter maxBytes: The maximum number of bytes this method is allowed to accumulate. 
+    /// - Returns: Bytes collected over time
+    public func bytes(upTo maxBytes: Int) async throws -> ByteBuffer {
+        let expectedBytes = self.headers.first(name: "content-length").flatMap(Int.init) ?? maxBytes
```

Comment:
This shouldn't just blindly trust the content-length: it should reject values of the content length above the `upTo` value.